pkg/plugins/connection/ssh.go

```diff
@@ -0,0 +1,74 @@
+package connection
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/nornir-automation/gornir/pkg/gornir"
+
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh"
+)
+
+// SSH is a Connection plugins that connects to device via ss using the golang.org/x/crypto/ssh
+// package. Current implementation only supports authentication with a password and has
+// ssh.InsecureIgnoreHostKey set
+type SSH struct {
+	Client *ssh.Client
+}
+
+// Close closes the connection
+func (s *SSH) Close() error {
+	return s.Client.Close()
+}
+
+// String implemente Stringer interface
+func (s SSH) String() string {
+	if s.Client == nil {
+		return "  - connection closed"
+	}
+	return "  - connection opened"
+}
+
+// SSHOpen is a Connection plugin that opens a connection with a device
+type SSHOpen struct {
+}
```

Comment:
I would add some placeholders here for config details. E.g.

```go
type SSHOpen struct {
	IgnoreHostKey bool
}
```

Reply:
Agreed.